src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -67,190 +68,186 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
+                    int overloadCount;
+
+                    // Select the overload with minimum number of parameters to display
+                    if (overloadMap.TryGetValue((containingNamespacename, symbol.Name, symbol.Arity > 0), out var currentValue))
                     {
-                        builder.Add(nameof(Object));
+                        bestSymbol = currentValue.bestSymbol.Parameters.Length > symbol.Parameters.Length ? symbol : currentValue.bestSymbol;
+                        overloadCount = currentValue.overloadCount + 1;
                     }
-                }
-            }
-        }
-
-        private static string GetReceiverTypeName(ITypeSymbol typeSymbol)
-        {
-            switch (typeSymbol)
-            {
-                case INamedTypeSymbol namedType:
-                    return namedType.MetadataName;
-
-                case IArrayTypeSymbol arrayType:
-                    var elementType = arrayType.ElementType;
-                    while (elementType is IArrayTypeSymbol symbol)
+                    else
                     {
-                        elementType = symbol.ElementType;
+                        bestSymbol = symbol;
+                        overloadCount = 1;
                     }
 
-                    var elementTypeName = GetReceiverTypeName(elementType);
-
-                    // We do not differentiate array of different kinds sicne they are all represented in the indices as "NonArrayElementTypeName[]"
-                    // e.g. int[], int[][], int[,], etc. are all represented as "int[]", whereas array of complex type such as T[] is "[]".
-                    return elementTypeName + FindSymbols.Extensions.ArrayReceiverTypeNameSuffix;
-
-                default:
-                    // Complex types are represented by "";
-                    return FindSymbols.Extensions.ComplexReceiverTypeName;
+                    overloadMap[(containingNamespacename, symbol.Name, symbol.Arity > 0)] = (bestSymbol, overloadCount);
+                }
             }
-        }
 
-        public static async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)> GetUnimportedExtensionMethodsInCurrentProcessAsync(
-            Document document,
-            int position,
-            ITypeSymbol receiverTypeSymbol,
-            ISet<string> namespaceInScope,
-            bool forceIndexCreation,
-            CancellationToken cancellationToken)
-        {
-            var counter = new StatisticCounter();
-            var ticks = Environment.TickCount;
+            // Then convert symbols into completion items
+            using var _3 = ArrayBuilder<SerializableImportCompletionItem>.GetInstance(out var itemsBuilder);
 
-            var indicesResult = await TryGetIndicesAsync(
-                document.Project, forceIndexCreation, cancellationToken).ConfigureAwait(false);
+            foreach (var (methodData, overloadSymbols) in overloadMap)
```

Comment:
can you not do:
```suggestion
            foreach (var ((containingNamespace, _, _), (bestSymbol, overloadCount)) in overloadMap)
```

?